### PR TITLE
Prayerslayer #134 snapshot source plugin

### PR DESCRIPTION
**Pull request by jbellmann**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="http://app.review.ninja/zalando-stups/fullstop/pull/161" target="_blank">ReviewNinja

File: fullstop-plugin-api/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                Region.getRegion(Regions.fromName(event.getEventData().getAwsRegion())));
+
+        DescribeInstanceAttributeRequest describeInstanceAttributeRequest = new DescribeInstanceAttributeRequest();
+        describeInstanceAttributeRequest.setInstanceId(instanceId);
+        describeInstanceAttributeRequest.setAttribute(USER_DATA);
+
+        DescribeInstanceAttributeResult describeInstanceAttributeResult;
+        describeInstanceAttributeResult = ec2Client.describeInstanceAttribute(describeInstanceAttributeRequest);
+
+        String userData = describeInstanceAttributeResult.getInstanceAttribute().getUserData();
```
Comment (mrandi) [inline fullstop-plugin-api/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java]: what happens if the instance is already down?

File: fullstop-plugins/fullstop-snapshot-source-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+    <build>
+		<plugins>
```
Comment (mrandi) [inline fullstop-plugins/fullstop-snapshot-source-plugin/pom.xml]: :sleeping: on :musical_keyboard: :stuck_out_tongue_winking_eye:

File: fullstop/pom.xml [hunk 3]
```diff
@@ -68,22 +68,22 @@
         <!-- START PLUGINS -->
         <dependency>
             <groupId>${project.groupId}</groupId>
-            <artifactId>fullstop-count-events-plugin</artifactId>
+            <artifactId>fullstop-ami-plugin</artifactId>
```
Comment (mrandi) [inline fullstop/pom.xml]: why did you move all dependencies? It's pretty difficult to do a review and check what is new and what was moved.

Comment (prayerslayer) [inline fullstop/pom.xml]: Alphabetical sorting. Will make future reviews and finding plugins easier.

File: fullstop-plugin-api/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java (hunk 1 above)
Comment (prayerslayer) [inline fullstop-plugin-api/src/main/java/org/zalando/stups/fullstop/events/UserDataProvider.java]: I guess the AWS SDK throws an exception. Has to be handled by the plugin.
